pkg/profileparser/profileparser.go

```diff
@@ -376,15 +375,13 @@ func parseProfileFromNode(profileRoot *xmldom.Node, pb *cmpv1alpha1.ProfileBundl
 	return nil
 }
 
-func getProductTypeAndName(root *xmldom.Node, defaultType cmpv1alpha1.ComplianceScanType, defaultName string) (cmpv1alpha1.ComplianceScanType, string) {
-	// Loop ourselves b/c we are only interested in direct children
-	for _, child := range root.Children {
-		if child.Name == "platform" {
-			return parseProductTypeAndName(child.GetAttributeValue("idref"), defaultType, defaultName)
-		}
-	}
+func getProductTypeAndName(root *xmlquery.Node, defaultType cmpv1alpha1.ComplianceScanType, defaultName string) (cmpv1alpha1.ComplianceScanType, string) {
+	p := root.SelectElement("xccdf-1.2:platform")
```

Comment:
So this part is tricky. See that getProducTypeAndName is used twice, this is IIRC because each profile can redefine the CPE. That's why the original code used to only check for direct children and not just search for any platform. You could do that (I think) by searching for `/platform`. But the thing is that I don't know if any profile we ship does this -- so maybe it's just additional complexity we don't need to have in the code. IIRC I was just coding by following the standard back then.

Reply:
We do this with `ocp4-cis-node`, where we say that applies for a node and not the platform. That's a good point! I need to verify that one.

Reply:
Seems to me like it parsed it correctly:
```
apiVersion: compliance.openshift.io/v1alpha1
description: This profile defines a baseline that aligns to the Center for Internet
  Security®&#xA;Red Hat OpenShift Container Platform 4 Benchmark™, V0.3, currently
  unreleased.&#xA;&#xA;This profile includes Center for Internet Security®&#xA;Red
  Hat OpenShift Container Platform 4 CIS Benchmarks™ content.&#xA;&#xA;Note that this
  part of the profile is meant to run on the Operating System that&#xA;Red Hat OpenShift
  Container Platform 4 runs on top of.&#xA;&#xA;This profile is applicable to OpenShift
  versions 4.6 and greater.
id: xccdf_org.ssgproject.content_profile_cis-node
kind: Profile
metadata:
  annotations:
    compliance.openshift.io/image-digest: pb-ocp4hx7kq
    compliance.openshift.io/product: redhat_openshift_container_platform_node_4
    compliance.openshift.io/product-type: Node
  creationTimestamp: "2021-03-23T06:52:56Z"
  generation: 1
  labels:
    compliance.openshift.io/profile-bundle: ocp4
  name: ocp4-cis-node
  namespace: openshift-compliance
```